addons/pos_loyalty/static/src/overrides/components/product_screen/order_summary/order_summary.js

```diff
@@ -62,34 +62,27 @@ patch(OrderSummary.prototype, {
      */
     _setValue(val) {
         const selectedLine = this.currentOrder.get_selected_orderline();
-        if (
-            !selectedLine ||
-            !selectedLine.is_reward_line ||
-            (selectedLine.is_reward_line && ["", "remove"].includes(val))
-        ) {
-            super._setValue(val);
-        }
         if (!selectedLine) {
             return;
         }
         if (selectedLine.is_reward_line && val === "remove") {
-            this.currentOrder.uiState.disabledRewards.add(selectedLine.reward_id);
-            const { couponCache } = this.pos;
-            const coupon = couponCache[selectedLine.coupon_id];
+            this.currentOrder.uiState.disabledRewards.add(selectedLine.reward_id.id);
+            const coupon = selectedLine.coupon_id;
             if (
                 coupon &&
                 coupon.id > 0 &&
-                this.currentOrder.uiState.codeActivatedCoupons.find((c) => c.code === coupon.code)
+                this.currentOrder.code_activated_coupon_ids.find((c) => c.code === coupon.code)
             ) {
-                delete couponCache[selectedLine.coupon_id];
-                this.currentOrder.uiState.codeActivatedCoupons.splice(
-                    this.currentOrder.uiState.codeActivatedCoupons.findIndex((coupon) => {
-                        return coupon.id === selectedLine.coupon_id;
-                    }),
-                    1
-                );
+                coupon.delete();
             }
         }
+        if (
+            !selectedLine ||
+            !selectedLine.is_reward_line ||
+            (selectedLine.is_reward_line && ["", "remove"].includes(val))
+        ) {
+            super._setValue(val);
+        }
```

Comment:
I don't really understand why you place this code here rather than at the beginning of the method